programs/ssl/ssl_client2.c

```diff
@@ -1327,11 +1330,17 @@ int main( int argc, char *argv[] )
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     if( opt.key_opaque != 0 )
     {
-        /* coming soon: load key to key slot */
+        if( ( ret = mbedtls_pk_wrap_as_opaque( &pkey, &key_slot,
+                                               PSA_ALG_SHA_256 ) ) != 0 )
```

Comment:
I would rather suggest defining it closer to the key or even get it from parsing and using the translating utility rather than hardcoding.

Reply:
In the long run I agree, but for the time being I'm going to consider it out of scope of this PR.

The main purpose was to make sure that opaque keys can be used for client authentication, ie that there wasn't any need to change code in the TLS module in order to adapt to opaque keys (same for your CSR generation PR, which made it clear that some change was needed in pkwrite, except here it turned out no change was needed). That purpose can be achieved with a hardcoded hash.

But yeah, I assume a subsequent PR is going to change that.